sdk/nodejs/x/automation/workspace.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ConfigMap, ConfigValue } from "./config";
+import { ProjectSettings } from "./projectSettings";
+import { StackSettings } from "./stackSettings";
+
+/**
+ * Workspace is the execution context containing a single Pulumi project, a program, and multiple stacks.
+ * Workspaces are used to manage the execution environment, providing various utilities such as plugin
+ * installation, environment configuration ($PULUMI_HOME), and creation, deletion, and listing of Stacks.
+ *
+ * @alpha
+ */
+export interface Workspace {
+    /**
+     * The working directory to run Pulumi CLI commands
+     */
+    readonly workDir: string;
+    /**
+     * The directory override for CLI metadata if set.
+     * This customizes the location of $PULUMI_HOME where metadata is stored and plugins are installed.
+     */
+    readonly pulumiHome?: string;
+    /**
+     * The secrets provider to use for encryption and decryption of stack secrets.
+     * See: https://www.pulumi.com/docs/intro/concepts/config/#available-encryption-providers
+     */
+    readonly secretsProvider?: string;
+    /**
+     *  The inline program `PulumiFn` to be used for Preview/Update operations if any.
+     *  If none is specified, the stack will refer to ProjectSettings for this information.
+     */
+    program?: PulumiFn;
+    /**
+     * Environment values scoped to the current workspace. These will be supplied to every Pulumi command.
+     */
+    envVars: { [key: string]: string };
```

Comment:
Any reason these aren't also `readonly`?

Reply:
Yup, they are mutable and can change throughout the lifecycle of a stack (or multiple stacks) used with the workspace.